src/components/ChoiceCollectionSelectInput/index.tsx

```diff
@@ -132,13 +137,30 @@ function ChoiceCollectionSelectInput<
         variables: optionsVariables,
     });
 
+    const handleChoiceAddSuccess = useCallback((newChoice: ChoiceCollectionType) => {
+        otherProps?.onChange(newChoice.id, name);
+
+        if (otherProps?.onOptionsChange) {
+            otherProps?.onOptionsChange((oldOptions) => (
+                [
+                    ...(oldOptions ?? []),
+                    newChoice,
+                ]
+            ));
```

Comment:
When newChoice is updated (not added), chances are the oldOptions might already have that option. Make sure you search and remove that before you add the newChoice(updated choice) in the list.